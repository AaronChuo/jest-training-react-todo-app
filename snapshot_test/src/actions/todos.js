import {
  ADD_TODO,
  DELETE_TODO,
  EDIT_TODO,
  COMPLETE_TODO,
  COMPLETE_ALL,
  CLEAR_COMPLETED,
} from 'constants/todos';

export const addTodo = (payload) => {
  return {
    type: ADD_TODO,
    payload,
  };
};

export const deleteTodo = (payload) => {
  return {
    type: DELETE_TODO,
    payload,
  };
};

export const editTodo = (payload) => {
  return {
    type: EDIT_TODO,
    payload,
  };
};

export const completeTodo = (payload) => {
  return {
    type: COMPLETE_TODO,
    payload,
  };
};

export const completeAll = () => {
  return {
    type: COMPLETE_ALL,
  };
};

export const clearCompleted = () => {
  return {
    type: CLEAR_COMPLETED,
  };
};
