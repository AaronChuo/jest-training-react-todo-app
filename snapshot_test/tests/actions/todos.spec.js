import * as actions from 'actions/todos';

describe('Todos actions', () => {
  it('should dispatch ADD_TODO action', () => {
    expect(actions.addTodo({})).toMatchSnapshot();
  });

  it('should dispatch DELETE_TODO action', () => {
    expect(actions.deleteTodo({})).toMatchSnapshot();
  });

  it('should dispatch EDIT_TODO action', () => {
    expect(actions.editTodo({})).toMatchSnapshot();
  });

  it('should dispatch COMPLETE_TODO action', () => {
    expect(actions.completeTodo({})).toMatchSnapshot();
  });

  it('should dispatch COMPLETE_ALL action', () => {
    expect(actions.completeAll()).toMatchSnapshot();
  });

  it('should dispatch CLEAR_COMPLETED action', () => {
    expect(actions.clearCompleted()).toMatchSnapshot();
  });
});
