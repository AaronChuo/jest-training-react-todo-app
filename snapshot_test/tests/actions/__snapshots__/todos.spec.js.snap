// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos actions should dispatch ADD_TODO action 1`] = `
Object {
  "payload": Object {},
  "type": "ADD_TODO",
}
`;

exports[`Todos actions should dispatch CLEAR_COMPLETED action 1`] = `
Object {
  "type": "CLEAR_COMPLETED",
}
`;

exports[`Todos actions should dispatch COMPLETE_ALL action 1`] = `
Object {
  "type": "COMPLETE_ALL",
}
`;

exports[`Todos actions should dispatch COMPLETE_TODO action 1`] = `
Object {
  "payload": Object {},
  "type": "COMPLETE_TODO",
}
`;

exports[`Todos actions should dispatch DELETE_TODO action 1`] = `
Object {
  "payload": Object {},
  "type": "DELETE_TODO",
}
`;

exports[`Todos actions should dispatch EDIT_TODO action 1`] = `
Object {
  "payload": Object {},
  "type": "EDIT_TODO",
}
`;
